From b268bd1c94912fbb5b490f9914bfcdd8fb467b6e Mon Sep 17 00:00:00 2001
From: Alon Bar-Lev <alon.barlev@gmail.com>
Date: Wed, 15 Aug 2012 09:46:42 +0300
Subject: [PATCH 4/5] support route metric

This adds optional ROUTE_METRIC environment variable to vpnc-script to
allow set non default routes and dns metrics.

Signed-off-by: Alon Bar-Lev <alon.barlev@gmail.com>
---
 vpnc-script | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/vpnc-script b/vpnc-script
index eb9749d..9a98628 100755
--- a/vpnc-script
+++ b/vpnc-script
@@ -238,7 +238,7 @@ if [ -n "$IPROUTE" ]; then
 		NETWORK="$1"
 		NETMASK="$2"
 		NETMASKLEN="$3"
-		$IPROUTE route replace "$NETWORK/$NETMASKLEN" dev "$TUNDEV"
+		$IPROUTE route replace "$NETWORK/$NETMASKLEN" dev "$TUNDEV" ${ROUTE_METRIC:+metric ${ROUTE_METRIC}}
 		$IPROUTE route flush cache
 	}
 
@@ -553,7 +553,7 @@ nameserver $i"
 		NEW_RESOLVCONF="$NEW_RESOLVCONF
 domain $CISCO_DEF_DOMAIN"
 	fi
-	echo "$NEW_RESOLVCONF" | ${RESOLVCONF} -a $TUNDEV
+	echo "$NEW_RESOLVCONF" | ${RESOLVCONF} ${ROUTE_METRIC:+-m ${ROUTE_METRIC}} -a $TUNDEV
 }
 
 restore_resolvconf_manager() {
-- 
1.8.1.5

